resteasy-jaxrs/src/main/java/org/jboss/resteasy/plugins/providers/sse/SseEventOutputImpl.java

```diff
@@ -174,9 +217,48 @@ protected void writeEvent(OutboundSseEvent event) throws IOException
          {
             if (event != null)
             {
+               //// Check media type?
                ByteArrayOutputStream bout = new ByteArrayOutputStream();
-               writer.writeTo(event, event.getClass(), null, new Annotation[]
-               {}, event.getMediaType(), null, bout);
+               MediaType mediaType = event.getMediaType();
+               boolean mediaTypeSet = event instanceof OutboundSseEventImpl ? ((OutboundSseEventImpl) event).isMediaTypeSet() : true;
+               if (mediaType == null || !mediaTypeSet)
+               {
+                  Object o = response.getOutputHeaders().getFirst("Content-Type");
+                  if (o != null)
+                  {
+                     if (o instanceof MediaType)
+                     {
+                        MediaType mt = (MediaType) o;
+                        String s = mt.getParameters().get(SseConstants.SSE_ELEMENT_MEDIA_TYPE);
+                        if (s != null)
+                        {
+                           mediaType = MediaType.valueOf(s);
+                        }
+                     }
+                  }
+                  else if (o instanceof String)
```

Comment:
@ronsigal This condition doesn't make sense. `if (o instanceof String)` statement is reachable only if `o == null` (see line 227). And if o == null,  `if (o instanceof String)` is evaluated to false. So ` if (o instanceof String)` is always evaluated as false, so lines 239-247 are useless now. Can you please check&&fix it?

Reply:
@marekkopecky, d'oh! It looks like I copy and pasted to the wrong place. Thanks.